docs/resources/client.example.hcl

```diff
@@ -6,10 +6,14 @@ client {
     ]
 }
 
-# plugin "docker" {
-#   config {
-#     auth {
-#       config = "/root/.docker/config.json"
-#     }
-#   }
-# }
+plugin "docker" {
+    config {
+        gc {
+            image_delay = "1m"
```

Comment:
> Then, the allocations will first redownload the image and finally start the runners. A potential side effect of this mechanism would be that images get updated / downloaded more often, depending on Nomad's scheduler.

What would be an appropriate timeout?

Reply:
Good question; I don't know. However, I would consider the following scenario to determine the delay time:

Let's assume we have two agents (A and B) running allocations of our execution environment and the system is being used actively. First, both agents have some allocations with image version 1. Due to the inactivity timeout, the last allocation is stopped on agent A (all other allocations are on agent B). Within the `image_delay`, I instruct Nomad to download a newer image version 2 and recreate the environments. I see two possibilities here:

1. The allocations on agent B get stopped and are recreated on agent B exclusively. I assume that all allocations will have the new image version 2.
2. The allocations on agent B get stopped and are recreated on agents A and B. As the `force_pull` option is included for all allocations, all allocations should use the new image 2.

This is no issue so far. However, what happens if we continue the experiment and assume the first outcome (all allocations updated on agent B). Now, a learner requests a new runner, and one from agent B is taken (as agent A doesn't have an allocation yet). Still, while being in the `image_delay`, Poseidon creates a new allocation to refill the prewarming pool size. This time, the allocation is created on agent A (due to Nomad's internal scheduler). 

My question: Which image version is used by the new allocation?